sockeye/decoder.py

```diff
@@ -105,51 +143,291 @@ def get_rnn_cells(self) -> List[mx.rnn.BaseRNNCell]:
         raise NotImplementedError()
 
 
-DecoderState = NamedTuple('DecoderState', [
+class TransformerDecoder(Decoder):
+    """
+    Transformer decoder as in Vaswani et al, 2017: Attention is all you need.
+    In training, computation scores for each position of the known target sequence are compouted in parallel,
+    yielding most of the speedup.
+    At inference time, the decoder block is evaluated again and again over a maximum length input sequence that is
```

Comment:
That was my first idea too. Its not trivial so I want to leave this out of scope of this PR. But obviously there are a couple of things that can be done to speed this up.
Overall its actually not that slow on a GPU :)